src/ui/view/settings/boxes.rs

```diff
@@ -0,0 +1,290 @@
+use iced::{Align, Column, Container, Length, Row};
+
+use crate::ui::{
+    component::{card, separation, text},
+    message::Message,
+};
+
+use crate::revaultd::config::Config;
+
+pub trait SettingsBox {
+    fn title(&self) -> &'static str;
+    fn description(&self) -> &'static str;
+    fn body<'a>(&self, config: &Config) -> Column<'a, Message>;
+    fn display<'a>(&self, config: &Config) -> Container<'a, Message> {
+        card::simple(Container::new(
+            Column::new()
+                .push(
+                    Row::new()
+                        .push(
+                            Container::new(
+                                Row::new()
+                                    .push(
+                                        Column::new()
+                                            .push(text::bold(text::simple(self.title())))
+                                            .push(text::small(self.description())),
+                                    )
+                                    .spacing(20),
+                            )
+                            .width(Length::Fill),
+                        )
+                        .spacing(20)
+                        .align_items(Align::Center),
+                )
+                .push(separation().width(Length::Fill))
+                .push(self.body(config))
+                .spacing(20),
+        ))
+        .width(Length::Fill)
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct SettingsBoxes {
+    pub general: GeneralBox,
+    pub bitcoind: BitcoindBox,
+    pub stakeholder_xpubs: StakeholderXpubsBox,
+    pub manager_xpubs: ManagerXpubsBox,
+    pub cosigner_keys: CosignerKeysBox,
+    pub manager: ManagerBox,
+    pub stakeholder: StakeholderBox,
```

Comment:
The user may be only a Stakeholder and may not have manager settings and
The user may be only a Manager and may not have stakeholder settings

Reply:
I was quite unsure if I should keep both as `Option` or not. The reason why I went with both as non-optional is that I'm checking [here](https://github.com/revault/revault-gui/pull/57/files/a855037fa4ef13932e9b088086c26486a0daf8f1#diff-1a0ad2f92d087268c81da488ce7bb082ac2f39c4f6d12dd292ce12ce27d83615R54) the role in order to display the correct box. Otherwise I could implement on SettingBoxes the functions `stakeholder()` and `manager()` and call `SettingBoxes::stakeholder()` or `SettingBoxes::manager()` depending on the context role, but I feel that this would make the code a bit bloated. What do you think?

Reply:
StakeholderBox and ManagerBox are empty structures anyway, that's why I didn't even bother with Option. The trouble comes when you call `display()` on the StakeholderBox and you're a Manager or viceversa...

Reply:
Ok :+1: , It's good for now. We may improve it once we start having feedbacks on usability